### ISSUE TITLE
redefined-outer-name not emitted for names in enclosing namespace

### ISSUE THREAD

**Issue by superbobry** [PyCQA/pylint#5678]
### Bug description

pylint does detect redefinitions in enclosing scope and instead goes directly to globals. So, the following example does not produce `redefined-outer-name`

```
def wrapper():
    x = 42
    def f(x):
        pass
```

whereas this one does

```
x = 42
def f(x):
    pass
```

If this is WAI, I think we should consider adjusting the documentation for `redefined-outer-name`. 

### Command used

```shell
N/A
```

### Pylint output

```shell
N/A
```

### Expected behavior

`redefined-outer-name` is emitted in both examples.

### Pylint version

```shell
pylint 2.8.3
astroid 2.5.6
Python 3.9.9 (main, Nov 16 2021, 10:24:31) 
[GCC 11.2.0]
```

**Comment by Pierre-Sassoulas** [comment]
Confirmed to still be an issue in 2.13.0-dev0
